atom/browser/atom_browser_context.cc

```diff
@@ -219,6 +105,15 @@ void AtomBrowserContext::RegisterPrefs(PrefRegistrySimple* pref_registry) {
   pref_registry->RegisterDictionaryPref(prefs::kDevToolsFileSystemPaths);
 }
 
+std::string AtomBrowserContext::GetUserAgent() const {
+  return user_agent_;
+}
+
+void AtomBrowserContext::OnMainRequestContextCreated(
+    brightray::URLRequestContextGetter* getter) {
+  getter->set_delegate(request_context_delegate_.get());
```

Comment:
Have changed the implementation a bit now, request_context_delegate_ is owned by `AtomBrowserContext` and the only reason it cannot be owned by `URLRequestContextGetter` is because of the cookie change notification. The getter will outlive the delegate and all the cross thread notifications in the delegate are protected by a `WeakPtr`. Also the destruction of the getter isn't signaled until `brightray::BrowserContext` is destroyed.